apps/studio/components/interfaces/Auth/AuthProvidersForm/ProviderForm.tsx

```diff
@@ -113,10 +116,35 @@ export const ProviderForm = ({ config, provider, isActive }: ProviderFormProps)
     )
   }
 
+  // Open or close the form based on the query parameter
+  useEffect(() => {
+    const isProviderInQuery =
+      typeof router.query.provider === 'string' &&
+      router.query.provider.toLowerCase() === provider.title.toLowerCase()
+    setOpen(isProviderInQuery)
+  }, [router.query.provider, provider.title])
+
+  // Update URL when form is closed via the Sheet's onOpenChange
+  const handleOpenChange = (isOpen: boolean) => {
+    if (!isOpen && router.query.provider === provider.title) {
+      // Remove provider query param when closed
+      const query = { ...router.query }
+      delete query.provider
+      router.replace({ pathname: router.pathname, query }, undefined, { shallow: true })
+    }
+  }
+
+  // Handle clicking on a provider in the list
+  const handleProviderClick = () => {
+    // Update URL with provider query param
+    const query = { ...router.query, provider: provider.title }
+    router.replace({ pathname: router.pathname, query }, undefined, { shallow: true })
+  }
+
```

Comment:
i'd opt to use `useUrlState` here actually, then it'll remove the need for all the manual router updates with its inbuilt `setParams`

Reply:
Good shout – I forgot about that hook!